crates/polars-core/src/frame/mod.rs

```diff
@@ -411,6 +411,29 @@ impl DataFrame {
         DataFrame { columns }
     }
 
+    /// Create a new `DataFrame` but does not check the length of the `Series`,
+    /// only check for duplicates.
+    ///
+    /// It is advised to use [Series::new](Series::new) in favor of this method.
+    ///
+    /// # Panic
+    /// It is the callers responsibility to uphold the contract of all `Series`
+    /// having an equal length, if not this may panic down the line.
+    pub fn new_no_length_checks(columns: Vec<Series>) -> PolarsResult<DataFrame> {
+        let mut names = PlHashSet::with_capacity(columns.len());
+        for column in &columns {
+            let name = column.name();
+            if !names.insert(name) {
+                // If insertion fails, it means the element is already in the set (duplicate)
+                polars_bail!(duplicate = name)
+            }
```

Comment:
This is a better implementation than the one in `new` - perhaps update there too? The insertion doesn't technically 'fail' though, so I would reword that comment (or just leave it out - the code is clear enough imo).